File: algorithms/sorting/bubble-sort/bubble_sort.jl
arr = [8,2,4,3]

function bubble_sort!(arr::Vector{Int})
    for i in 1:length(arr)
        for j in 1:length(arr) - i
            if arr[j] > arr[j+1]
                arr[j], arr[j+1] = arr[j+1], arr[j]
            end
        end
    end
end

bubble_sort!(arr)

File: algorithms/sorting/bubble-sort/bubble_sort.rs
let mut arr = [8,2,4,3];

fn bubble_sort(arr: &mut [i32])
{
    for i in 0..arr.len()
    {
        for j in 0..arr.len() - 1 - i
        {
            if arr[j] > arr[j+1]
            {
                arr.swap(j, j+1);
            }
        }
    }
}

fn main()
{
    bubble_sort(&mut arr);
}